src/pandas_openscm/index_manipulation.py

```diff
@@ -714,3 +717,188 @@ def update_levels_from_other(
     res = pd.MultiIndex(levels=levels, codes=codes, names=names)
 
     return res
+
+
+def create_level_from_collection(
+    level: str, value: Collection[Any]
+) -> tuple[pandas.Index[Any], list[int]]:
+    """
+    Create new level and corresponding codes.
+
+    Parameters
+    ----------
+    level
+        Name of the level to create
+
+    value
+        Values to use to create the level
+
+    Returns
+    -------
+    :
+        New level and corresponding codes
+    """
+    new_level: pandas.Index[Any] = pd.Index(value, name=level)
+    if not new_level.has_duplicates:
+        # Fast route, can just return new level and codes from level we mapped from
+        return value, list(range(len(value)))
```

Comment:
```suggestion
        return new_level, np.arange(len(value))
```

mypy was complaining because you were returning `value` (which is a collection) rather than `new_level` (which is a pd.Index)